src/db/schemas/settlement.model.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  pgEnum,
+  pgTable,
+  real,
+  timestamp,
+  varchar,
+} from "drizzle-orm/pg-core";
+import { createInsertSchema, createSelectSchema } from "drizzle-zod";
+import type { z } from "zod";
+
+import { splitTypeArr } from "@/common/enums";
+
+import groupModel from "./group.model";
+import userModel from "./user.model";
+
+export const splitTypeEnum = pgEnum("splitType", splitTypeArr);
+
+const settlementModel = pgTable("settlement", {
+  id: varchar({ length: 60 })
+    .$defaultFn(() => Bun.randomUUIDv7())
+    .primaryKey()
+    .notNull(),
+  payerId: varchar({ length: 60 })
+    .notNull()
+    .references(() => userModel.id, {
+      onDelete: "cascade",
+      onUpdate: "cascade",
+    }),
+  receiverId: varchar({ length: 60 })
+    .notNull()
+    .references(() => userModel.id, {
+      onDelete: "cascade",
```

Comment:
I think these foreign constraints are not needed here.
We do not wish to delete the record if a user is deleted or a group is deleted.

Reply:
I think right. I was also bit confused while adding them.